runtime/marshaler_registry.go

```diff
@@ -31,7 +32,11 @@ func MarshalerForRequest(mux *ServeMux, r *http.Request) (inbound Marshaler, out
 	}
 
 	for _, contentTypeVal := range r.Header[contentTypeHeader] {
-		if m, ok := mux.marshalers.mimeMap[contentTypeVal]; ok {
+		contentType, _, err := mime.ParseMediaType(contentTypeVal)
+		if err != nil {
+			continue
+		}
```

Comment:
Maybe we could log to the `grpclog` when this errors? I think we do that in other parts of the codebase. It could be the source of confusion to users (like I'm guessing it was to you before this patch).